### PR TITLE
fix(route): 修复风之动漫

### PR DESCRIPTION
**Pull request by zytomorrow**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/fffdm/manhua/2?limit=2
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
修复风之动漫漫画获取，使用最新风之动漫API，不再解析网页

### CODE REVIEW

File: lib/v2/fzdm/manhua/manhua.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        const chapterTitle = data.title;
+        const pics = data.cont;
+        return { comicTitle, chapterTitle, pics };
+    });
+};
+
+module.exports = async (ctx) => {
+    const id = ctx.params.id;
+    const count = ctx.query.limit || 99999;
+    const cdn = `https://p${ctx.params.cdn || 5}.fzacg.com`;
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:22]: https://github.com/DIYgod/RSSHub/security/advisories/GHSA-64wp-jh9p-5cg2

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/fzdm/manhua/manhua.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    const response = await got.get(`${host}/api/manhua/${id}`);
+    const data = response.data;
+
+    const chapter_detail = await Promise.all(
+        data.mhlist.splice(0, count).map(async (item) => {
+            const pic_content = await get_pic(id, item.url, ctx.cache);
+            return {
+                title: pic_content.chapterTitle,
+                description: art(path.resolve(__dirname, '../templates/manhua.art'), { pic: pic_content.pics, cdn }),
+                link: `${host}/api/manhua/${id}/${item.url}`,
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:40]:
```suggestion
                link: `${host}/${id}/${item.url}/`,
```

File: lib/v2/fzdm/manhua/manhua.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                comicTitle: pic_content.comicTitle,
+            };
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:42]: Please cache the whole returned result instead.

File: lib/v2/fzdm/manhua/manhua.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        })
+    );
+    ctx.state.data = {
+        title: '风之动漫 - ' + chapter_detail[0].comicTitle,
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:46]: Maybe use `data.mhinfo.latest.title` instead as it looks like it's more reliable

File: lib/v2/fzdm/manhua/manhua.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+const get_pic = (id, chapter, caches) => {
+    const url = `${host}/api/manhua/${id}/${chapter}`;
+    return caches.tryGet(url, async () => {
+        const response = await got.get(url);
+        const data = response.data;
+        const comicTitle = data.mhinfo.title;
+        const chapterTitle = data.title;
+        const pics = data.cont;
+        return { comicTitle, chapterTitle, pics };
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:15]: The API also provides pubDate in `mh.time`

File: lib/v2/fzdm/manhua/manhua.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    const cdnNum = ctx.params.cdn || 5;
+    let cdn = '';
+    if (!isNaN(parseFloat(cdnNum)) && 1 <= parseFloat(cdnNum) && parseFloat(cdnNum) <= 5) {
+        cdn = `https://p${cdnNum}.fzacg.com`;
+    } else {
+        cdn = `https://p5.fzacg.com`;
+    }
+
+    // 获取漫画清单
+    const response = await got.get(`${host}/api/manhua/${id}`);
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:31]:
```suggestion
    const response = await got(`${host}/api/manhua/${id}`);
```

File: lib/v2/fzdm/manhua/manhua.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
+const got = require('@/utils/got');
+const domain = 'manhua.fffdm.com';
+const host = `https://${domain}`;
+const { art } = require('@/utils/render');
+const path = require('path');
+
+const get_pic = (id, chapter, caches) => {
+    const url = `${host}/api/manhua/${id}/${chapter}`;
+    return caches.tryGet(url, async () => {
+        const response = await got.get(url);
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:10]:
```suggestion
        const response = await got(url);
```

File: lib/v2/fzdm/radar.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'fffdm.com': {
+        _name: '风之动漫',
+        manhua: [
+            {
+                title: '风之动漫',
+                docs: 'https://docs.rsshub.app/anime.html#feng-zhi-dong-man-feng-zhi-dong-man',
+                source: ['/*'],
+                target: (params, url) => `/fzdm/manhua/${new URL(url).pathname.split('/')[1]}/`,
```
Comment (TonyRL) [inline lib/v2/fzdm/radar.js:9]:
```suggestion
                source: ['/:id'],
                target: '/fzdm/manhua/:id',
```

File: lib/v2/fzdm/manhua/manhua.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        return { comicTitle, chapterTitle, pics };
+    });
+};
+
+module.exports = async (ctx) => {
+    const id = ctx.params.id;
+    const count = ctx.query.limit || 99999;
+    const cdnNum = ctx.params.cdn || 5;
+    let cdn = '';
+    if (!isNaN(parseFloat(cdnNum)) && 1 <= parseFloat(cdnNum) && parseFloat(cdnNum) <= 5) {
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:24]: Please use `parseInt`

File: lib/v2/fzdm/manhua/manhua.js (hunk 4 above)
Comment (zytomorrow) [reply lib/v2/fzdm/manhua/manhua.js:46]: image
the data is wrong sometime.

File: lib/v2/fzdm/radar.js (hunk 8 above)
Comment (zytomorrow) [reply lib/v2/fzdm/radar.js:9]: This change can't play in content page,just like `https://manhua.fffdm.com/2/1072/`

Comment (TonyRL) [reply lib/v2/fzdm/radar.js:9]: You can use
```js
source: ['/:id', '/:id/*'],
target: '/fzdm/manhua/:id',
```
instead

Comment (zytomorrow) [reply lib/v2/fzdm/radar.js]: thx

File: lib/v2/fzdm/manhua/manhua.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
+const got = require('@/utils/got');
+const domain = 'manhua.fffdm.com';
+const host = `https://${domain}`;
+const { art } = require('@/utils/render');
+const path = require('path');
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:7]:
```suggestion
```

File: lib/v2/fzdm/manhua/manhua.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+const get_pic = async (url) => {
+    const response = await got(url);
+    const data = response.data;
+    return {
+        comicTitle: data.mhinfo.title,
+        chapterTitle: data.title,
+        pics: data.cont,
+        pubDate: timezone(parseDate(data.mh.time, 'YYYY-MM-DDTHH:mm:ss.SSSZ'), +8),
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:16]:
```suggestion
        pubDate: parseDate(data.mh.time),
```
No need to adjust timezone for ISO 8601 timestring
[more info](https://stackoverflow.com/a/36798830)

File: lib/v2/fzdm/manhua/manhua.js [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    // 获取漫画清单
+    const response = await got(`${host}/api/manhua/${id}`);
+    const data = response.data;
+
+    const chapter_detail = await Promise.all(
+        data.mhlist.splice(0, count).map(async (item) => {
+            const url = `${host}/api/manhua/${id}/${item.url}`;
+            return await ctx.cache.tryGet(url, async () => {
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:38]:
```suggestion
        data.mhlist.splice(0, count).map((item) => {
            const url = `${host}/api/manhua/${id}/${item.url}`;
            return ctx.cache.tryGet(url, async () => {
```

File: lib/v2/fzdm/manhua/manhua.js [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+                const picContent = await get_pic(url);
+                return {
+                    title: picContent.chapterTitle,
+                    description: art(path.resolve(__dirname, '../templates/manhua.art'), { pic: picContent.pics, cdn }),
```
Comment (TonyRL) [inline lib/v2/fzdm/manhua/manhua.js:42]:
```suggestion
                    description: art(path.join(__dirname, '../templates/manhua.art'), { pic: picContent.pics, cdn }),
```

**Pull request merged**